### PR TITLE
Animal form

### PR DESCRIPTION
**Pull request by pablo-grande**
Beware travelers! here be dragons.

A few things to keep in mind before merging this branch:
1. The unknown breed is supposed to be existing in DB (ref: #1)
https://github.com/petconnection/backend/blob/9723c6c3ed820546202d770c9c2e2372cd250404/backoffice/views.py#L26
2. The form is ugly AF, needs a few upgrades (ref: #17)
3. You may have some troubles to enter and see the changes in local because of this:
https://github.com/petconnection/backend/blob/9723c6c3ed820546202d770c9c2e2372cd250404/backoffice/views.py#L9
We do not have a proper login yet (ref: #14) so... yeah, proably it can be bypassed by accessing first to `/admin` and logging there and then go to "View Site"

### CODE REVIEW

**Comment by pablo-grande** [comment]
4. About error messages on form submitting
https://github.com/petconnection/backend/blob/9723c6c3ed820546202d770c9c2e2372cd250404/backoffice/static/scripts.js#L27
This just gets a message provided by
https://github.com/petconnection/backend/blob/9723c6c3ed820546202d770c9c2e2372cd250404/backoffice/views.py#L53
Providing only hints if the form has some kind of validation error. This means if an exception occurs it would return nothing, so maybe we should keep that in mind and think a way of tackle this issue

File: backoffice/views.py [hunk 1]
```diff
@@ -1,12 +1,55 @@
 from django.shortcuts import render
 from django.contrib.auth.decorators import login_required
-from api.models import Animal, Entity
+from api.models import *
```
Comment (Nosk14) [inline backoffice/views.py]: Wildcard imports should be avoided. 
https://www.python.org/dev/peps/pep-0008/#imports

**Comment by pablo-grande** [comment]
5. No image uploading of the animal was implemented (sorry). Updating #17

File: backoffice/views.py [hunk 2]
```diff
@@ -1,12 +1,55 @@
...
-from api.models import Animal, Entity
+import api.models as models
+from .forms import AnimalForm
+from django.utils import timezone
+from django.http import JsonResponse as json
 
 
 @login_required(login_url='login')
 def home(request):
+    from django.contrib.auth import authenticate
```
Comment (Nosk14) [inline backoffice/views.py]: Why is this import here? And where is it used?

Comment (pablo-grande) [reply backoffice/views.py]: This is the remainings of an old bypass for the login method and to avoid a `@login_required`. It should (and will be) removed.

**Comment by pablo-grande** [comment]
Hello there! :wave: 
There was a MAJOR refactor of the form, check out this beauty:


- [x] Looks way better. There is now a 3 level form, with separated contextual information.
- [x] Upper right buttons toogle between views, if we are adding an animal they will change into:


- [x] Better viewing of errors if form failed
- [x] Change species selection for a dropdown select: http://creativeit.github.io/getmdl-select/
- [x] Edit button on animal cards should load an edit view with same form with populated fields
- [x] "Home" title should also be dynamic, telling the user where is he at every moment, with a `{% block second_title %}` maybe?

**Comment by pablo-grande** [comment]
The new push changed color scheme in order to highlight errors

**Comment by pablo-grande** [comment]
Sorry for the double commiting, I tried to close the issue in a cool way but I misspelled the number, I force pushed but it seems it didn't work well. And the funny thing is it is not completed! :sweat: 

- [x] Pic upload

- [x] Weight!

**Comment by pablo-grande** [comment]
- [x] Delete feature

File: api/models.py [hunk 3]
```diff
@@ -43,10 +43,18 @@ class Animal(models.Model):
...
-        breed_field = self.breed_field
-        species_field = breed_field.species_field
-        return "{name} {animal_id}: {breed} {species}".format(name=name, animal_id=self.id, breed=breed_field, species=species_field)
-
+        breed = self.breed_field
+        species = breed.species_field
+        return "{} {}: {} {}".format(name, self.id, breed, species)
+
+    @property
+    def medical_record(self):
```
Comment (Nosk14) [inline api/models.py]: This property is not necessary. You can use `animal.medicalrecord` instead. It is an autogenerated property from the OneToOne relationship

File: backoffice/static/scripts.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  var file_text = $(".file_text");
+
+  file_input.change(function(){
+    var filename = $(this).val().split('fakepath\\')[1];
+    file_text.text(filename);
+  });
+
+
+  //asign semi-random color to a card
+  var colors = ['teal', 'pink', 'purple', 'deep-purpe', 'indigo', 'blue', 'light-blue',
```
Comment (Nosk14) [inline backoffice/static/scripts.js]: There is a typo in `deep-purple`. The L is missing

**Review by Nosk14 — CHANGES_REQUESTED** [review]
There are two strange behaviors in the clear button.
- Not all fields are cleared.
- After a clear, if I want to check a check button again, I need to click it twice.
